user_guide_src/source/testing/overview.rst

```diff
@@ -60,24 +60,31 @@ Testing Your Application
 PHPUnit Configuration
 =====================
 
-The framework has a ``phpunit.xml.dist`` file in the project root. This controls unit
-testing of the framework itself. If you provide your own ``phpunit.xml``, it will
-over-ride this.
+In your CodeIgniter project root, there is the ``phpunit.xml.dist`` file. This
+controls unit testing of your application. If you provide your own ``phpunit.xml``,
+it will over-ride this.
 
-Your ``phpunit.xml`` should exclude the ``system`` folder, as well as any ``vendor`` or
-``ThirdParty`` folders, if you are unit testing your application.
+By default, test files are placed under the **tests** directory in the project root.
 
 The Test Class
 ==============
 
-In order to take advantage of the additional tools provided, your tests must extend ``CIUnitTestCase``. All tests
-are expected to be located in the **tests/app** directory by default.
+In order to take advantage of the additional tools provided, your tests must extend
+``CIUnitTestCase``.
 
-To test a new library, **Foo**, you would create a new file at **tests/app/Libraries/FooTest.php**:
+There are no rules for how test files must be placed. However, we recommend that
+you establish placement rules in advance so that you can quickly understand where
+the test files are located.
+
+In this document, we will place the test files corresponding to the classes in
+the **app** directory in the **tests/app** directory. To test a new library,
+**app/Libraries/Foo.php**, you would create a new file at
+**tests/app/Libraries/FooTest.php**:
```

Comment:
Not a big fan of the `tests/app/...` folder structure. But this might be only my preference, and since we already are using it in the docs, I won't be pushing for a change.

Reply:
Yes, since we already use it in the docs, I just use it here.
> All tests are expected to be located in the tests/app directory by default.
https://www.codeigniter.com/user_guide/testing/overview.html#the-test-class